Patch Tuesday for December 2024

BillyONeal opened the pull request:
* Update Azure-CLI to 2.67.0.
* Update Visual Studio to 17.12.3.
* Update Android base image to noble-20241118.1 and fix incorrect PowerShell registry.
* Deindex casadi, also extracted as independent PR https://github.com/microsoft/vcpkg/pull/42773 .
* Change ryu to a CMake build system and add a test port for it.

BillyONeal committed:
Update Azure-CLI.

BillyONeal committed:
Update Visual Studio to 17.12.3.

BillyONeal committed:
Update Windows pool.

BillyONeal committed:
Update Android image.

dg0yt commented:
Reminder: We need more linux packages in the Android image, to bootstrap host dependencies.
'libwayland-dev' 'wayland-protocol'
Maybe system OpenJDK (untested)

And eventually the SDK from Google.
https://github.com/microsoft/vcpkg/pull/42080/files#

BillyONeal committed:
Remove nonexistent python2 package from Android Noble image generation.

BillyONeal commented:
> Reminder: We need more linux packages in the Android image, to bootstrap host dependencies.

I think I've bitten off enough for a monthly update here getting the Ubuntu version finally not ancient but will look to do that shortly after if this works.


@cenit FYI The cause of the explosion before was `python2` not existing on Ubuntu Noble.

BillyONeal committed:
[ryu] Replace build system with a CMake one, and add a test port.

BillyONeal committed:
PASSING, REMOVE FROM FAIL LIST: openfbx:arm-neon-android

BillyONeal committed:
[casadi] #41649 appears to have been merged in error, casadi/casadi#3896 needs to land before this can be in the curated registry. Moreover, problems in the portfile were skipped in ci.baseline.txt rather than a supports expression. We are happy to bring this port back as soon as that issue is fixed :)

Additionally, this port appears to cause a hang in CI attempting to update to Visual Studio 17.12.3:

BillyONeal committed:
Merge remote-tracking branch 'BillyONeal/deindex-casadi' into patch-tuesday-december-2024

BillyONeal committed:
Version DB

BillyONeal committed:
Disable saucer due to VS 17.12 compiler bug.
